libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -336,9 +348,11 @@ private void bind() {
   }
 
   private void initViewModels() {
-    locationViewModel = ViewModelProviders.of(this).get(LocationViewModel.class);
-    routeViewModel = ViewModelProviders.of(this).get(RouteViewModel.class);
-    navigationViewModel = ViewModelProviders.of(this).get(NavigationViewModel.class);
+    if (getContext() instanceof FragmentActivity) {
```

Comment:
Should we throw the `ClassCastException` here?

Reply:
Yes - see https://github.com/mapbox/mapbox-navigation-android/pull/346/files#r144303130.